dependencies.yaml

```diff
@@ -448,6 +508,25 @@ dependencies:
         packages:
           # Allow runtime version to float up to patch version
           - pyarrow>=16.1.0,<16.2.0a0
+  run_libcudf:
```

Comment:
We should name this `depends_on_libcudf`. This doesn't express the run dependencies _of_ libcudf, it expresses a run dependency _on_ libcudf.

Reply:
Ah, I see `depends_on_libcudf` below. In that case, this section is probably superfluous and can be deleted.

Reply:
Agreed, thanks!

I've removed this `run_libcudf` completely and used a `depends_on_libcudf` everywhere, similar to the other changes from https://github.com/rapidsai/cudf/pull/16611.